woocommerce/woocommerce-ios
# Shipping Labels: Customs form UI

Part of #4687 

# Description
This PR aims to implement part of the customs form UI since code changes are getting massive. The rest of the form will be updated in the next PR.

# Changes
- Added new property for package name in `ShippingLabelCustomsForm`. This is actually an internal implementation, but I figure it allows for cleaner setup of the form details so I'm leaving it this way - feel free to let me know if you have better suggestion for this 🙇‍♀️ 
- New reusable SwiftUI views:
  - `CollapsibleView`: view that can be collapsed / expanded to show / hide contents. This is to prepare for multi-package support and also item details later on.
  - `SelectionList`: A list for selecting items.
- Updated some reusable views:
  - `TitleAndToggleRow`: flexible height to allow more than 1 line for title and option for sub-headline styling.
  - `SelectableItemRow`: options for compact mode and trailing checkmark.
- Updated UI for customs form list and input views and view models.
- Miscellanous:
  - Also fixed `customsFormRequired` to remove deprecated check for US domestic territories (ref: p1628513294009100/1628066200.023400-slack-C026HH75ECA).
  - Updated `ShippingLabelFormViewModel` to fix empty customs form list.

# Screenshots
| Light Mode | Dark Mode |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/128847048-1c33ec92-425f-499d-863d-0a671fec7a4a.png" width=400 /> | <img src="https://user-images.githubusercontent.com/5533851/128847080-72bd368a-6ff5-455d-a03c-bab71956ee0a.png" width=400 /> |
| <img src="https://user-images.githubusercontent.com/5533851/128847155-51eee7c4-a45c-4903-bb9d-f3b3cc4e8f19.png" width=400 /> | <img src="https://user-images.githubusercontent.com/5533851/128847197-7be65cac-36cc-4e4f-a907-e8f0b5088a4d.png" width=400 /> |

# Testing steps
1. Make sure that your test store is eligible for shipping labels (has WCShip plugin installed).
2. Create an order with international shipping (origin country different from destination country).
3. Navigate to Orders tab, select the new order and select Create Shipping Label.
4. Skip through Ship From, Ship To, Package Details to configure Customs.
5. Notice that the Customs screen show correct name for your selected package.
6. Notice that the default contents type is Merchandise. Tapping on Contents Type row should navigate to list of options, and selecting any option and tapping done should redirect to Customs form view with the correct selected contents type.

Regression tests for updated reusable views:
- Selected package list should still look as expected.
- Toggle row in Payment Method screen should still look as expected.

Update release notes:
